tests/helpers/test_entity_component.py

```diff
@@ -554,32 +554,70 @@ async def generate_response(
     response_data = await hass.services.async_call(
         DOMAIN,
         "hello",
-        service_data={"entity_id": entity.entity_id, "some": "data"},
+        service_data={"some": "data"},
+        target={"entity_id": [entity.entity_id]},
         blocking=True,
         return_response=True,
     )
-    assert response_data == {"response-key": "response-value"}
+    assert response_data == {f"{DOMAIN}.entity": {"response-key": "response-value"}}
 
 
 async def test_register_entity_service_response_data_multiple_matches(
     hass: HomeAssistant,
 ) -> None:
-    """Test asking for service response data but matching many entities."""
+    """Test asking for service response data and matching many entities."""
     entity1 = MockEntity(entity_id=f"{DOMAIN}.entity1")
     entity2 = MockEntity(entity_id=f"{DOMAIN}.entity2")
 
     async def generate_response(
         target: MockEntity, call: ServiceCall
     ) -> ServiceResponse:
-        raise ValueError("Should not be invoked")
+        return {"response-key": "response-value"}
```

Comment:
Maybe include the entity data in the response just to ensure each response is associated is not getting jumbled across entities? e.g.  f"response-value-{target.entity_id}"

Reply:
Done